src/containers/collectibles/solCollectibleHelpers.ts

```diff
@@ -0,0 +1,188 @@
+import {
+  Collectible,
+  CollectibleMediaType
+} from 'containers/collectibles/types'
+import {
+  SolanaNFT,
+  SolanaNFTPropertiesFile
+} from 'services/solana-client/types'
+import { Chain } from 'store/token-dashboard/slice'
+import { Nullable } from 'utils/typeUtils'
+
+import { getFrameFromGif } from './ethCollectibleHelpers'
+
+type SolanaNFTMedia = {
+  collectibleMediaType: CollectibleMediaType
+  url: string
+  frameUrl: Nullable<string>
+}
+
+const nftGif = async (nft: SolanaNFT): Promise<Nullable<SolanaNFTMedia>> => {
+  const gifFile = nft.properties.files?.find(
+    file => typeof file === 'object' && file.type === 'image/gif'
+  )
+  if (gifFile) {
+    const url = (gifFile as SolanaNFTPropertiesFile).uri
+    const frameUrl = await getFrameFromGif(url, nft.name)
+    return { collectibleMediaType: CollectibleMediaType.GIF, url, frameUrl }
+  }
+  return null
+}
+
+const nftVideo = async (nft: SolanaNFT): Promise<Nullable<SolanaNFTMedia>> => {
+  const files = nft.properties.files
+  // should we restrict video file extensions here?
+  // MP4, MOV, GLB
+  // GLTF??
+  // https://github.com/metaplex-foundation/metaplex/blob/81023eb3e52c31b605e1dcf2eb1e7425153600cd/js/packages/web/src/views/artCreate/index.tsx#L318
+  // DO WE CARE ABOUT VR NFTs??
+  const videoFile = files?.find(
+    file => typeof file === 'object' && file.type.includes('video')
+  ) as SolanaNFTPropertiesFile
+  // https://github.com/metaplex-foundation/metaplex/blob/397ceff70b3524aa0543540584c7200c79b198a0/js/packages/web/src/components/ArtContent/index.tsx#L107
```

Comment:
this link already broke I think. Try: https://github.com/metaplex-foundation/metaplex/blob/194a82f35ecc5178c9595ea1dbbdc0bdf4ec42b4/js/packages/web/src/components/ArtContent/index.tsx#L107

Reply:
did it? im still able to access